service/history/mutableStateBuilder.go

```diff
@@ -259,45 +263,18 @@ func newMutableStateBuilder(
 	return s
 }
 
-func newMutableStateBuilderWithVersionHistories(
+func newMutableStateBuilderFromDB(
 	shard shard.Context,
 	eventsCache events.Cache,
 	logger log.Logger,
 	namespaceEntry *cache.NamespaceCacheEntry,
-	startTime time.Time,
-) *mutableStateBuilder {
-
-	s := newMutableStateBuilder(shard, eventsCache, logger, namespaceEntry)
-	// start time should be set for workflow timeout calculation
-	// NOTE: workflow reset case, this start time is the reset time
-	s.executionInfo.StartTime = timestamp.TimePtr(startTime)
-	s.executionInfo.VersionHistories = versionhistory.NewVersionHistories(&historyspb.VersionHistory{})
-	return s
-}
-
-func (e *mutableStateBuilder) CloneToProto() *persistencespb.WorkflowMutableState {
-	ms := &persistencespb.WorkflowMutableState{
-		ActivityInfos:       e.pendingActivityInfoIDs,
-		TimerInfos:          e.pendingTimerInfoIDs,
-		ChildExecutionInfos: e.pendingChildExecutionInfoIDs,
-		RequestCancelInfos:  e.pendingRequestCancelInfoIDs,
-		SignalInfos:         e.pendingSignalInfoIDs,
-		SignalRequestedIds:  convert.StringSetToSlice(e.pendingSignalRequestedIDs),
-		ExecutionInfo:       e.executionInfo,
-		ExecutionState:      e.executionState,
-		NextEventId:         e.hBuilder.NextEventID(),
-		BufferedEvents:      e.bufferEventsInDB,
-		Checksum:            e.checksum,
-	}
-
-	return proto.Clone(ms).(*persistencespb.WorkflowMutableState)
-}
-
-// TODO unify Load and StartTransaction
-func (e *mutableStateBuilder) Load(
 	state *persistencespb.WorkflowMutableState,
```

Comment:
Probably worth to rename `state` to `mutableState` or `ms` here. Or leave it for the future refactorings.

Reply:
i have one PR to be send out in few minutes, i can address it there